src/odemis/acq/calibration.py

```diff
@@ -358,6 +372,22 @@ def apply_spectrum_corrections(data, bckg=None, coef=None):
     return data
 
 
+def remove_nan(l: list, data: model.DataArray) -> Tuple[list, model.DataArray]:
```

Comment:
this could use a more descriptive name, maybe something like `filter_nan_from_data`